datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -231,21 +241,34 @@ def list_all_nodes_with_details(
         )
         .limit(NODE_LIST_MAX)
     )  # Very high limit as a safeguard
-    nodes = session.exec(nodes_query).all()
-
-    if len(nodes) == NODE_LIST_MAX:  # pragma: no cover
+    results = [
+        NodeIndexItem(name=row[0], display_name=row[1], description=row[2], type=row[3])
+        for row in session.exec(nodes_query).all()
+    ]
+    if len(results) == NODE_LIST_MAX:  # pragma: no cover
         _logger.warning(
             "%s limit reached when returning all nodes, all nodes may not be captured in results",
             NODE_LIST_MAX,
         )
-    accessable_nodes = validate_access_nodes(
-        validate_access,
-        access.ResourceRequestVerb.BROWSE,
-        current_user,
-        nodes,
-    )
-
-    return [node.current for node in accessable_nodes]
+    approvals = [
+        approval.access_object.name
+        for approval in validate_access_requests(
+            validate_access,
+            current_user,
+            [
+                access.ResourceRequest(
+                    verb=access.ResourceRequestVerb.BROWSE,
+                    access_object=access.Resource(
+                        name=row.name,
+                        resource_type=access.ResourceType.NODE,
+                        owner="",
```

Comment:
Is this empty because we're not yet storing owner on a node?

Reply:
Yes that's exactly right. I'm planning to follow up with a PR that stores owner. Sort of a side discussion but I have some opinions on the usefulness of an owner field. I think we should have "created_by" here instead. To me, "ownership" is something that's implied by who created the node plus an analysis of the activity/history.

Reply:
Yeah, +1 to created_by, which we can derive from the node history